plugin/src/main/groovy/com/novoda/buildproperties/internal/DefaultExceptionFactory.groovy

```diff
@@ -0,0 +1,34 @@
+package com.novoda.buildproperties.internal
+
+class DefaultExceptionFactory implements ExceptionFactory {
+
+    private final String propertiesSetName
+    private final ConsoleRenderer consoleRenderer
+    private String additionalMessage
+
+    DefaultExceptionFactory(String propertiesSetName, ConsoleRenderer consoleRenderer = new ConsoleRenderer()) {
+        this.propertiesSetName = propertiesSetName
+        this.consoleRenderer = consoleRenderer
+    }
+
+    @Override
+    void setAdditionalMessage(String additionalMessage) {
+        this.additionalMessage = additionalMessage
+    }
+
+    @Override
+    Exception fileNotFound(File file) {
+        BuildPropertiesException.from("Properties set 'buildProperties.$propertiesSetName' is defined upon ${consoleRenderer.asClickableFileUrl(file)}, but the file does not exist.", formattedAdditionalMessage())
```

Comment:
"defined upon" sounds weird but the sentence makes sense.

```groovy
"Unable to create properties set 'buildProperties.$propertiesSetName': could not find ${consoleRenderer.asClickableFileUrl(file)}."
```

sounds more idiomatic for me.